llvm/test/TableGen/escape-dependency-filenames.td

```diff
@@ -0,0 +1,38 @@
+// RUN: rm -rf %t; split-file %s %t
+
+//--- normal-file.td
+class NormalClass {}
+
+//--- file with spaces.td
+class SpaceClass {}
+
+//--- file#with#hash.td
+class HashClass {}
+
+//--- file$with$dollar.td
+class DollarClass {}
+
+//--- file with escape\ before spaces.td
```

Comment:
Can you also add a test for backslash _not_ before a space, since that is handled differently right?

Reply:
Thanks -- done in edec47b352e71c467615ff1d95ef1c60bb7324df.